### ISSUE TITLE
Explicit nil checks both ways don't narrow value to never

### ISSUE THREAD

**Issue by Luka967** [EmmyLuaLs/emmylua-analyzer-rust#1140]
When checking for truthiness both ways:
```lua
local target --- @type string
local source --- @type string
if not source then return end
if source then return end
target = source
```
At assignment `source` has reported type `never` and warning on assignment is properly emitted. It is technically unreachable code too, but it's beyond the scope of this issue.

When `nil` checking explicitly:
```lua
local target --- @type string
local source --- @type string
if source ~= nil then return end
if source == nil then return end
target = source
```
At assignment `source` has reported type `string` but it is expected for `source` to be `never`. Likewise no warning is emitted.

emmylua_ls `v0.23.2`
VSCode `1.1.126`
LS integration on VSCode, `tangzx.emmylua`, version `0.9.39`
